app/controllers/authentication.rb

```diff
@@ -0,0 +1,38 @@
+# error 301 do
+  
+# end
+
+
+before '/*' do 
+  authorize_or_redirect unless request.path_info == '/login' or request.path_info == '/logout'
```

Comment:
I love that you are trying up the requirement of authorization on certain routes but this is a bit brittle. Can you think of a way to make this more flexible? Maybe a blacklist instead of a whitelist?